backend/src/main/java/de/amos/apachepulsarui/service/TopicService.java

```diff
@@ -41,24 +41,42 @@ public class TopicService {
 
     private final PulsarAdmin pulsarAdmin;
 
-    private final MessageService messageService;
-
     private final ConsumerService consumerService;
 
+    @Cacheable("topics.allForTopics")
     public List<TopicDto> getAllForTopics(List<String> topics) {
         return topics.stream()
                 .filter(this::exists)
                 .map(topic -> TopicDto.create(topic, getTopicStats(topic)))
                 .toList();
     }
 
+    @Cacheable("topics.allForNamespace")
     public List<TopicDto> getAllForNamespaces(List<String> namespaces) {
         return namespaces.stream()
                 .map(this::getByNamespace)
                 .flatMap(topics -> getAllForTopics(topics).stream())
                 .toList();
     }
 
+    @Cacheable("topics.detail")
+    public TopicDetailDto getTopicDetails(String topicName) throws PulsarApiException {
+        return TopicDetailDto.create(
+                topicName,
+                getTopicStats(topicName),
+                getOwnerBroker(topicName),
+                getSchemasOfTopic(topicName)
+        );
+    }
+
+    public void createNewTopic(String topic) throws PulsarApiException {
```

Comment:
We canprobably delete that right? since we shouldn't do writing changes

Reply:
Yes indeed 👍